src/button.tsx

```diff
@@ -151,8 +151,22 @@ const Button = React.forwardRef<TouchableElement, ButtonProps & {type: ButtonTyp
     const {loadingText} = props;
     const isSubmitButton = !!props.submit;
     const isFormSending = formStatus === 'sending';
+    const [isOnPressPromiseResolving, setIsOnPressPromiseResolving] = React.useState(false);
+
+    const onPressCallback = React.useCallback(
+        (e: React.MouseEvent<HTMLElement, MouseEvent>) => {
+            if (props.onPress) {
+                const result = props.onPress(e);
+                if (result) {
+                    setIsOnPressPromiseResolving(true);
+                    result.finally(() => setIsOnPressPromiseResolving(false));
+                }
+            }
+        },
+        [props]
```

Comment:
I think this would be enough:
```suggestion
        [props.onPress]
```